platform/broadcom/sonic-platform-modules-dell/s6100/scripts/s6100_platform.sh

```diff
@@ -45,27 +24,36 @@ if [[ "$1" == "init" ]]; then
 
     systemctl start --no-block s6100-ssd-upgrade-status.service
 
-    is_fast_warm=$(cat /proc/cmdline | grep SONIC_BOOT_TYPE | wc -l)
-
-    if [[ "$is_fast_warm" == "1" ]]; then
-        systemctl start --no-block s6100-i2c-enumerate.service
+    case "$(cat /proc/cmdline)" in
+        *SONIC_BOOT_TYPE=warm*)
+            TYPE='warm'
+            ;;
+        *SONIC_BOOT_TYPE=fastfast*)
+            TYPE='fastfast'
+            ;;
+        *SONIC_BOOT_TYPE=fast*|*fast-reboot*)
+            TYPE='fast'
+            ;;
```

Comment:
@sujinmkang do we need to add soft reboot here too?

Reply:
@yxieca ,will add "SONIC_BOOT_TYPE=soft" for the soft reboot case.